### PR TITLE
fix(TDI-42709): add Array supports

### PR DESCRIPTION
**Pull request by Voievidko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-42709

**What is the new behavior?**
Add support of JSON arrays

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/28/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/28/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/29/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/29/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arvoievidko%2FTDI-42709_JsonArray_support_for_couhcbase)

File: components/components-couchbase/src/main/java/org/talend/components/couchbase/runtime/CouchbaseWriter.java [hunk 1]
```diff
@@ -108,7 +109,13 @@ public JsonObject createHierarchicalJson(Schema schema, IndexedRecord record, in
                 JsonObject innerJson = JsonObject.fromJson(value.toString());
                 jsonObject.put(fieldName, innerJson);
             } catch (Exception e) {
-                jsonObject.put(fieldName, value);
+                try {
+                    JsonArray jsonArray = JsonArray.fromJson(value.toString());
+                    jsonObject.put(fieldName, jsonArray);
+                } catch (Exception e2) {
+                    // This mean it's not JSON object
+                    jsonObject.put(fieldName, value);
```
Comment (ypiel-talend) [inline components/components-couchbase/src/main/java/org/talend/components/couchbase/runtime/CouchbaseWriter.java]: Maybe add a test to check this use case which should add escape character if I well understand.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/39/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/39/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Arvoievidko%2FTDI-42709_JsonArray_support_for_couhcbase)

**Pull request merged**